actors/builtin/multisig/multisig_actor.go

```diff
@@ -385,23 +402,30 @@ func (a Actor) approveTransaction(rt vmr.Runtime, txnID TxnID, txn *Transaction)
 
 	// add the caller to the list of approvers
 	rt.State().Transaction(&st, func() interface{} {
+		ptx, err := adt.AsMap(adt.AsStore(rt), st.PendingTxns)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load pending transactions")
+
 		// update approved on the transaction
 		txn.Approved = append(txn.Approved, rt.Message().Caller())
-		if err := st.putPendingTransaction(adt.AsStore(rt), txnID, txn); err != nil {
+		if err := ptx.Put(txnID, txn); err != nil {
 			rt.Abortf(exitcode.ErrIllegalState, "failed to put transaction for approval: %v", err)
 		}
+
+		st.PendingTxns, err = ptx.Root()
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to flush pending transactions")
+
 		return nil
 	})
 
 	return executeTransactionIfApproved(rt, st, txnID, txn)
 }
 
-func (a Actor) getTransaction(rt vmr.Runtime, st State, txnID TxnID, proposalHash []byte, checkHash bool) *Transaction {
+func (a Actor) getTransaction(rt vmr.Runtime, ptx *adt.Map, txnID TxnID, proposalHash []byte, checkHash bool) *Transaction {
```

Comment:
Nit: this should not be a method, but a free function.

Reply:
This is done.